pypesto/petab/importer.py

```diff
@@ -740,7 +757,9 @@ def create_problem(
             x_names=x_ids,
             x_scales=x_scales,
             x_priors_defs=prior,
-            startpoint_method=self.create_startpoint_method(),
+            startpoint_method=self.create_startpoint_method(
+                x_ids=np.delete(x_ids, x_fixed_indices)
+            ),
```

Comment:
This will not work with models that have hierarchical parameters and `ESTIMATE==0` parameters (the issue is in the order of removing parameters from x_ids). 

In case of Boehm with `['Epo_degradation_BaF3', 'k_exp_hetero', 'k_exp_homo', 'k_imp_hetero', 'k_imp_homo', 'k_phos', 'ratio', 'sd_pSTAT5A_rel', 'sd_pSTAT5B_rel', 'sd_rSTAT5A_rel', 'specC17']` parameters, the ones that are hierarchical are `'sd_pSTAT5A_rel', 'sd_pSTAT5B_rel', 'sd_rSTAT5A_rel'` and these will be removed in these lines [here](https://github.com/ICB-DCM/pyPESTO/blob/9a754573a7b77d30d5dc1f67a8dc1be6c29f1640/pypesto/petab/importer.py#L704-L713). So then the 
`x_ids=['Epo_degradation_BaF3', 'k_exp_hetero', 'k_exp_homo', 'k_imp_hetero', 'k_imp_homo', 'k_phos', 'ratio', 'specC17']`. 

But then in when trying to call `np.delete(x_ids, x_fixed_indices)` it will give index error, because `x_fixed_indices` are `[6, 10]`, but the length has been reduced to 8 when the hierarchical ones were removed. 

So probably the easiest fix is using `x_ids = [x for x in x_ids if x not in self.petab_problem.x_fixed_ids]`. Or changing order of the deletions, but taking care that `x_scales` has to be made for fixed parameters too (probably).

Reply:
Hm, I don't think I can follow.

Around the line you referenced, `x_ids` is updated to only hold the outer parameters,  `x_fixed_indices` are updated to refer to this shortened list.
Later, for the sampling, we only need the outer parameters that are not fixed, which is what the `delete` is taking care of.

So, specifically, I don't think

> because x_fixed_indices are [6, 10], but the length has been reduced to 8

is true after `x_fixed_indices = list(x_ids.index, self.petab_problem.x_fixed_ids))`

Reply:
That's true. I missed the line where `x_fixed_indices` are updated.